AwsCloudIdProvider.cs

```diff
@@ -114,6 +114,22 @@ public async Task<string> GetCloudIdAsync()
                 return SignRequest(accessKey, secretKey, securityToken);
             }
 
+            // if failed, try to take from environment
+            // Apperantly lambda functions use env vars and not profiles
+             try {
+                var profileCreds = new Amazon.Runtime.EnvironmentVariablesAWSCredentials();
+                var creds = await profileCreds.GetCredentialsAsync();
+                var accessKey = creds.AccessKey;
+                var secretKey = creds.SecretKey;
+                var securityToken = creds.Token;
+                if (accessKey != "" && secretKey != "") {
```

Comment:
are you not validating securityToken? is it valid if empty?

Reply:
Yes, in some cases it comes empty. The signing code handles the case where it is.